cci_nepal/analysis/Data Analysis of Hill Data.py

```diff
@@ -0,0 +1,653 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pandas as pd
+import numpy as np
+
+import warnings
+import logging
+
+from matplotlib import pyplot as plt
+import seaborn as sns
+
+import cci_nepal
+from cci_nepal.getters.real_data import get_real_data as grd
+from cci_nepal.pipeline.real_data import data_manipulation as dm
+from cci_nepal.pipeline.real_data import nfri_list_file as nlf
+
+# get_ipython().run_line_magic('matplotlib', 'inline')
+plt.rcParams["figure.figsize"] = [10, 6]
+
+warnings.filterwarnings("ignore")
+
+pd.set_option("display.max_columns", None)
+pd.options.display.max_rows = 4000
+# get_ipython().run_line_magic('load_ext', 'nb_black')
+
+
+project_dir = cci_nepal.PROJECT_DIR
+logging.info("Logging the project directory below:")
+logging.info(project_dir)
+
+
+# #### Reading the data ####
+
+df = grd.read_csv_file(f"{project_dir}/inputs/data/real_data/Full_Consent_Hill.csv")
+
+
+# #### Dropping columns that are not related to Features (like introduction, notes, etc.)
+
+# In[5]:
+
+
+columns_to_drop = [
+    0,
+    1,
+    2,
+    3,
+    4,
+    7,
+    18,
+    31,
+    41,
+    43,
+    44,
+    56,
+    57,
+    73,
+    74,
+    75,
+    76,
+    77,
+    78,
+    79,
+    80,
+    81,
+    82,
+    83,
+    84,
+    85,
+]
+
+
+# In[6]:
+
+
+df.drop(df.columns[columns_to_drop], axis=1, inplace=True)
+
+
+# #### Missing Data ####
+
+# In[7]:
+
+
+df.isnull().sum().sort_values(ascending=False)
+
+
+# Mostly non voluntary questions where 0 was the expected answer are the missing value features. Thus, filling in the missing values with 0 in the step below.
+
+# In[8]:
+
+
+df.fillna(0, inplace=True)
+
+
+# #### Feature Creation ####
+
+# In[9]:
+
+
+df.insert(2, "Total_Members", df.iloc[:, 2:22].sum(axis=1))
+
+
+# In[10]:
+
+
+df.insert(3, "Total_Male", df.iloc[:, 3:13].sum(axis=1))
```

Comment:
As mentioned previously I think the key to not leaking in information on HH size into all of these features (anything that 'number of' eg males, females, children, income generating, is to transform them into percentages. Once they are percentages you can look at how these percentages correlate with the other features and the predictor values*. This will also help in the creation of the more high level features such as 'majority male' because depending on the distribution there may be a better way to cut it.  

*On the exploratory work you are doing, its probably worth putting aside a test set sooner rather than later so we don't accidentally feed information we learn from the dataset as a whole into the model. I know we are still waiting for all the data, can we put a random 20% (if thats the size we are cutting it off) aside already? We might need to take the sample in a way that ensures we are not not only oversampling from the majority district though. Or if we are only looking at hill data maybe we take 10% a put it aside now then take the other 10% from plain when we have the full data? Basically my point is we need to be very careful not to be influenced by the test data in our feature selection.